### PR TITLE
Types for reefjs

### PR DESCRIPTION
**Pull request by shockdevv**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/reefjs/reefjs-tests.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            return props.todos.length;
+        },
+    },
+});
+
+const app3 = new Reef('#app', {
+    store,
+    template: (props) => {
+        return `
+			<h1>${props.heading}</h1>
```
Comment (danger-public) [inline types/reefjs/reefjs-tests.ts:101]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

File: types/reefjs/reefjs-tests.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        },
+    },
+});
+
+const app3 = new Reef('#app', {
+    store,
+    template: (props) => {
+        return `<h1>${props.heading}</h1>
+        <ul>
+			${props.todos
```
Comment (danger-public) [inline types/reefjs/reefjs-tests.ts:102]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

**Comment by shockdevv** [comment]
What's wrong with my config file? It should be fine.

**Review by andrewbranch — APPROVED** [review]

**Comment by shockdevv** [comment]
Ready to merge

**Pull request merged**